### DISCUSSION TITLE
Search & open files in a specific set of directories recursively using mini.pick

### DISCUSSION THREAD

**Discussion by tachmonite37** [nvim-mini/mini.nvim#1369] [Q&A] [answered]
I'm trying to figure out a way to open/search files in a specific set of directories recursively. For example, C:\Program Files\Double Commander\doc , C:\Users\Anthony\Documents\01.20.SC.DOTS\01.23.SC.AUTH C:\Users\Anthony\Downloads\12.40.SC.LIT\12.43.LTST
using mini.pick.
Something similar to :

keymap("n", "<leader>fr", function() require("mini.extra").pickers.oldfiles() end, { noremap = true, silent = true, desc = 'Find RECENT File' })

I would prefer to easily/simply add to the set of directories using a bookmarks file. 

BTW. I'm using ONLY mini.nvim plugins and I'm loving it. I like the idea of using one set of ideas/philosophy to learn neovim (...and  little lua for that matter). I can say that I have certainly cured plugin hopping and information overload trying to understand how different plugin works. This has definitely sped up my learning curve. THANKS A LOT for your great work @echasnovski .

**Comment by echasnovski** [comment, marked as answer]
Thanks for kind words!



> I'm trying to figure out a way to open/search files in a specific set of directories recursively. For example, C:\Program Files\Double Commander\doc , C:\Users\Anthony\Documents\01.20.SC.DOTS\01.23.SC.AUTH C:\Users\Anthony\Downloads\12.40.SC.LIT\12.43.LTST
> using mini.pick.

Yes, it is possible. All built-in pickers can be given a `source.cwd` ("current working directory") as a field of its `opts` argument.

If you plan to do it frequently and interactively, here is [an example](https://github.com/echasnovski/mini.nvim/blob/c8922aef8207137e66c80acdb9523668599ba62a/doc/mini-pick.txt#L1214-L1219) of how to modify built-in `files` registry entry to accept `cwd`. With this you can do something like `:Pick files cwd='C:\\Program\ Files\\...'` (not escaping of `\` and spaces).

> I would prefer to easily/simply add to the set of directories using a bookmarks file.

That would be a bit tricky if you want to interactively modify those bookmarks. Otherwise it can be done by having a single list of bookmarks and a picker with `choose` spawning `builtin.files()` picker. Something like this:

```lua
MiniPick.registry.bookmark_dirs = function()
  -- Modify list as you want. Prefer full paths (see `:h fnamemodify()` maybe).
  local dirs = {
    '~/.config/nvim',
    '~/.local'
  }
  local choose = function(item)
    vim.schedule(function() MiniPick.builtin.files(nil, { source = { cwd = item } }) end)
  end
  return MiniPick.start({ source = { items = dirs, name = 'Bookmark dirs', choose = choose } })
end
```

Now you can do something like `:Pick bookmark_dirs`.
Or via mapping: `vim.keymap.set('n', '<Leader>fd', '<Cmd>Pick bookmark_dirs<CR>', { desc = 'Find in Directories' })`.

And I'd strongly recommend using `<Leader>f` prefix (and setting Leader key to space) for this type of mappings. The `f` key is very useful in Normal mode on its own.

**Reply by tachmonite37** [reply]
Fantastic! Thank you @echasnovski . This is exactly what I was looking for. I'll add it to my config, study it, and test it out for a while. I'll definitely leave some feedback after testing.
